src/Maestro/tests/Maestro.ScenarioTests/TestHelpers.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Maestro.ScenarioTests
+{
+    public static class TestHelpers
+    {
+        public static async Task<string> RunExecutableAsync(ITestOutputHelper testOutput, string executable, params string[] args)
+        {
+            testOutput.WriteLine($"{executable} {string.Join(" ", args.Select(a => $"\"{a}\""))}");
+            var output = new StringBuilder();
+
+            void ProcessDataReceived(object sender, DataReceivedEventArgs e)
+            {
+                if (e.Data != null)
+                {
+                    output.AppendLine(e.Data);
+                    testOutput.WriteLine(e.Data);
+                }
+            }
+
+            var psi = new ProcessStartInfo(executable)
+            {
+                RedirectStandardError = true,
+                RedirectStandardOutput = true
+            };
+            foreach (var arg in args)
+            {
+                psi.ArgumentList.Add(arg);
+            }
+
+            using var process = new Process
+            {
+                StartInfo = psi
+            };
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            process.OutputDataReceived += ProcessDataReceived;
```

Comment:
>ProcessDataReceived [](start = 42, length = 19)

I find that reading the 2 streams leads to a slightly easier to read code (and more correct).

In this particular case, the events might trigger after the process has exited, which is almost certainly going to lead to strange or incorrect behavior.

It's a bit tricky to handle right, but you need something vaguely like:
```
var readTask = process.Output.ReadLine();
var errorTask = process.Error.ReadLine();
var exitTask = tcs.Task;

var list = List<Task>{read, error, exit};
while(list.Count != 0) {
  var done = await Task.WhenAny(list);
  list.Remove(done);
  if (done == exit) {
    continue;
  }
  if (done == readTask) {
    builder.Add("OUTPUT" + await readTask);
    list.Add(readTask = process.Output.ReadLine();
    continue;
  }
  // that but for error
}
```

It's complicated, but it assures that A) both streams are completely drained, so you don't get weirdly truncated output  B) the streams are interleaved correctly and safely (there is no guarantee that the output and error events are triggered thread safe).